srcCpp/Infrastructure_pro.cxx

```diff
@@ -45,14 +53,19 @@ PerftestClock &PerftestClock::getInstance()
     return instance;
 }
 
-unsigned long long PerftestClock::getTimeUsec()
+unsigned long long PerftestClock::getTime()
 {
+  #ifndef RTI_PERFTEST_NANO_CLOCK
     clock->getTime(clock, &clockTimeAux);
     RTINtpTime_unpackToMicrosec(
             clockSec,
             clockUsec,
             clockTimeAux);
     return clockUsec + 1000000 * clockSec;
+  #else
+    clock_gettime(CLOCK_MONOTONIC, &timeStruct);
```

Comment:
Check for error:
```
RETURN VALUE
       clock_gettime(), clock_settime(), and clock_getres() return 0 for success, or -1 for failure (in which case errno is set appropriately).
```

Reply:
I will leave this open, as I would not know what to do with that error and I am hesitant to have to do that extra assignment and checking.